File: python/contador_desc_sync.py
import time

N = 1_000_000


def countdown(n: int):
    while n > 0:
        n -= 1


if __name__ == "__main__":
    inicio = time.perf_counter()
    countdown(N)
    fin = time.perf_counter()

    t_ejecucion = fin - inicio

    print(f"Tiempo de ejecución: {t_ejecucion:.6f} segundos")

File: clang/contador_desc_sync.c
#include <stdio.h>
#include <time.h>

#define N 1000000


void countdown(int n) {
    while(n > 0) {
        n--;
    }
}

int main(void) {
       
    clock_t inicio = clock();

    countdown(N);

    clock_t fin = clock();

    double t_ejecucion = (double)(fin - inicio) / CLOCKS_PER_SEC;

    printf("Tiempo de ejecucion (secuencial): %.6f segundos\r\n", t_ejecucion);

    return 0;
}
